shared/javascripts/context_messenger.js

```diff
@@ -471,12 +471,20 @@ if (Privly === undefined) {
   SafariAdapter.prototype.setListener = function (callback) {
     if (this.getContextName() === 'BACKGROUND_SCRIPT') {
       safari.application.addEventListener("message", function(payload) {
-        callback(payload);
+        if (typeof payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+        } else {
+          callback(payload);
+        }
       }, true);
     }
     if (this.getContextName() === 'CONTENT_SCRIPT') {
       safari.self.addEventListener("message", function(payload) {
-        callback(payload);
+        if (typeof payload.name !== "undefined" && payload.name === "privlyMessage") {
+          callback(payload.message);
+        } else {
+          callback(payload);
```

Comment:
The else statement is called when the message is sent from a `postMessage()` function.